dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultQueryPlanner.java

```diff
@@ -302,6 +305,29 @@ else if ( !params.getFilterPeriods().isEmpty() )
         return queries;
     }
 
+    private DataQueryParams removeOffsetPeriodsIfNotNeeded( DataQueryParams params )
```

Comment:
Could we move this method to a utility class, add Javadoc and add unit test? E.g. `PeriodOffsetUtils`.

Reply:
Done + tests